### PR TITLE
rename `requirements-txt` and `requirements-txt-dev`

### PR DESCRIPTION
**Pull request by fpgmaas**
Rename `requirements-txt` and `requirements-txt-dev` to `requirements-file` and `requirements-file-dev`

**PR Checklist**

-   [x] A description of the changes is added to the description of this PR.
-   [x] If there is a related issue, make sure it is linked to this PR.
-   [x] If you've fixed a bug or added code that should be tested, add tests!
-   [x] Documentation in `docs` is updated

**Description of changes**

This PR proposes to rename `requirements-txt` and `requirements-txt-dev` to `requirements-files` and `requirements-files-dev`. If users still use the old parameters, they will be shown a deprecation warning:

> Warning: In an upcoming release, support for the `--requirements-txt` command-line option and the `requirements_txt` configuration parameter will be discontinued. Instead, use `--requirements-files` or `requirements_files` under the `[tool.deptry]` section in pyproject.toml.

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ To determine the project's dependencies, _deptry_ will scan the directory it is
     - development dependencies from `dev-dependencies.txt` and `dependencies-dev.txt`, if any exist
 
 _deptry_ can be configured to look for `pip` requirements files with other names or in other directories.
-See [Requirements txt](#requirements-txt) and [Requirements txt dev](#requirements-txt-dev).
+See [Requirements txt](#requirements-file) and [Requirements txt dev](#requirements-file-dev).
```
Comment (mkniewallner) [inline docs/usage.md:39]:
```suggestion
See [Requirements file](#requirements-file) and [Requirements file dev](#requirements-file-dev).
```

File: python/deptry/cli.py [hunk 2]
```diff
@@ -274,8 +292,8 @@ def deptry(
         ignore_notebooks=ignore_notebooks,
         ignore=ignore,
         per_rule_ignores=per_rule_ignores,
-        requirements_txt=requirements_txt,
-        requirements_txt_dev=requirements_txt_dev,
+        requirements_file=requirements_txt if requirements_txt else requirements_file,
+        requirements_file_dev=requirements_txt_dev if requirements_txt_dev else requirements_file_dev,
```
Comment (mkniewallner) [inline python/deptry/cli.py:296]: nit:
```suggestion
        requirements_file=requirements_txt or requirements_file,
        requirements_file_dev=requirements_txt_dev or requirements_file_dev,
```

File: python/deptry/deprecate/requirements_txt.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+from __future__ import annotations
+
+REQUIREMENTS_TXT_DEPRECATION_MESSAGE = (
+    "Warning: In an upcoming release, support for the `--requirements-txt` command-line "
+    "option and the `requirements_txt` configuration parameter will be discontinued. "
+    "Instead, use `--requirements-file` or `requirements_txt` under the `[tool.deptry]` "
```
Comment (mkniewallner) [inline python/deptry/deprecate/requirements_txt.py:6]:
```suggestion
    "Instead, use `--requirements-file` or `requirements_file` under the `[tool.deptry]` "
```

File: python/deptry/deprecate/requirements_txt.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    "section in pyproject.toml."
+)
+
+REQUIREMENTS_TXT_DEV_DEPRECATION_MESSAGE = (
+    "Warning: In an upcoming release, support for the `--requirements-txt-dev` command-line "
+    "option and the `requirements_txt_dev` configuration parameter will be discontinued. "
+    "Instead, use `--requirements-file-dev` or `requirements_txt_dev` under the `[tool.deptry]` "
```
Comment (mkniewallner) [inline python/deptry/deprecate/requirements_txt.py:13]:
```suggestion
    "Instead, use `--requirements-file-dev` or `requirements_file_dev` under the `[tool.deptry]` "
```

**Review by mkniewallner — COMMENTED** [review]
A few nit comments, but this mostly looks good to me. Just wondering if we should not name the parameters `requirements-files` and `requirements-files-dev` since they are lists.

**Comment by fpgmaas** [comment]
> A few nit comments, but this mostly looks good to me. Just wondering if we should not name the parameters `requirements-files` and `requirements-files-dev` since they are lists.

Good idea. I was doubting about this a bit, since `deptry . --requirements-files requirements.txt` feels less intuitive than `deptry . --requirements-file requirements.txt, and most projects likely have a single requirements file. But on the other hand, it is correct since it accepts a list. So let's change it to plural.

File: docs/usage.md [hunk 5]
```diff
@@ -36,7 +36,7 @@ To determine the project's dependencies, _deptry_ will scan the directory it is
     - development dependencies from `dev-dependencies.txt` and `dependencies-dev.txt`, if any exist
 
 _deptry_ can be configured to look for `pip` requirements files with other names or in other directories.
-See [Requirements txt](#requirements-txt) and [Requirements txt dev](#requirements-txt-dev).
+See [Requirements file](#requirements-files) and [Requirements file dev](#requirements-files-dev).
```
Comment (mkniewallner) [inline docs/usage.md:39]:
```suggestion
See [Requirements files](#requirements-files) and [Requirements files dev](#requirements-files-dev).
```

File: docs/usage.md [hunk 6]
```diff
@@ -228,40 +228,40 @@ ignore_notebooks = true
 deptry . --ignore-notebooks
 ```
 
-#### Requirements txt
+#### Requirements files
 
-List of [`pip` requirements files](https://pip.pypa.io/en/stable/user_guide/#requirements-files) that contain the source dependencies.
+List of [`pip` requirements files](https://pip.pypa.io/en/stable/user_guide/#requirements-filess) that contain the source dependencies.
```
Comment (mkniewallner) [inline docs/usage.md:233]:
```suggestion
List of [`pip` requirements files](https://pip.pypa.io/en/stable/user_guide/#requirements-files) that contain the source dependencies.
```

File: docs/usage.md [hunk 7]
```diff
@@ -228,40 +228,40 @@ ignore_notebooks = true
...
 ```shell
-deptry . --requirements-txt requirements.txt,requirements-private.txt
+deptry . --requirements-files requirements.txt,requirements-private.txt
 ```
 
-#### Requirements txt dev
+#### Requirements files dev
 
-List of [`pip` requirements files](https://pip.pypa.io/en/stable/user_guide/#requirements-files) that contain the source development dependencies.
+List of [`pip` requirements files](https://pip.pypa.io/en/stable/user_guide/#requirements-filess) that contain the source development dependencies.
```
Comment (mkniewallner) [inline docs/usage.md:251]:
```suggestion
List of [`pip` requirements files](https://pip.pypa.io/en/stable/user_guide/#requirements-files) that contain the source development dependencies.
```

File: python/deptry/dependency_getter/requirements_files.py [hunk 8]
```diff
@@ -16,37 +16,37 @@
...
             )
         )
 
         dev_dependencies = list(
             itertools.chain(
                 *(
-                    self._get_dependencies_from_requirements_file(file_name)
-                    for file_name in self._scan_for_dev_requirements_files()
+                    self._get_dependencies_from_requirements_files(file_name)
+                    for file_name in self._scan_for_dev_requirements_filess()
```
Comment (mkniewallner) [inline python/deptry/dependency_getter/requirements_files.py:33]:
```suggestion
                    for file_name in self._scan_for_dev_requirements_files()
```

File: python/deptry/dependency_getter/requirements_files.py [hunk 9]
```diff
@@ -16,37 +16,37 @@
...
                 )
             )
         )
 
         return DependenciesExtract(dependencies, dev_dependencies)
 
-    def _scan_for_dev_requirements_files(self) -> list[str]:
+    def _scan_for_dev_requirements_filess(self) -> list[str]:
```
Comment (mkniewallner) [inline python/deptry/dependency_getter/requirements_files.py:40]:
```suggestion
    def _scan_for_dev_requirements_files(self) -> list[str]:
```

File: python/deptry/dependency_getter/requirements_files.py [hunk 10]
```diff
@@ -16,37 +16,37 @@
...
+        Check if any of the files passed as requirements_files_dev exist, and if so; return them.
         """
-        dev_requirements_files = [file_name for file_name in self.requirements_txt_dev if file_name in os.listdir()]
-        if dev_requirements_files:
-            logging.debug("Found files with development requirements! %s", dev_requirements_files)
-        return dev_requirements_files
+        dev_requirements_filess = [file_name for file_name in self.requirements_files_dev if file_name in os.listdir()]
+        if dev_requirements_filess:
+            logging.debug("Found files with development requirements! %s", dev_requirements_filess)
+        return dev_requirements_filess
```
Comment (mkniewallner) [inline python/deptry/dependency_getter/requirements_files.py:47]:
```suggestion
        dev_requirements_files = [file_name for file_name in self.requirements_files_dev if file_name in os.listdir()]
        if dev_requirements_files:
            logging.debug("Found files with development requirements! %s", dev_requirements_files)
        return dev_requirements_files
```

File: docs/usage.md (hunk 6 above)
Comment (fpgmaas) [reply docs/usage.md]: Oops, replacement error... My bad.

**Review by mkniewallner — APPROVED** [review]

**Pull request merged**